spec/support/acceptance/curl_command.rb

```diff
@@ -0,0 +1,48 @@
+module Serverspec
+  module Type
+    class Curl < Command
+      def response_code
+        m = /Response-Code: (?<code>\d+)/.match(stderr)
+        return 0 unless m
+        m[:code].to_i
+      end
+
+      def body
+        command_result.stdout
+      end
+
+      def body_as_json
+        MultiJson.load(body)
+      end
+
+      private
+
+      def curl_command
+        command = "curl --silent --write-out '%{stderr}Response-Code: %{response_code}\\n' '#{@name}'"
```

Comment:
curl 7.70.0 introduces `--write-out %{json}` to output everything as JSON but even Fedora 32 doesn't ship that yet